### ISSUE TITLE
LaTeX: let footnotes contain backrefs links, especially when multiply referred to

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#8848]
It is already possible by raw directives hack to let footnotes of a chapter be printed only on last page of chapter (using ``\savenotes/\spewnotes`` but perhaps this is not optimal as it puts all chapter in a TeX scoping group; there is perhaps some extra latex package not conflicting with own Sphinx latex code which does this but it is not probable as the balancing with hyperref is delicate and we are only interested in solutions with working clickable footnote markers). 

Anyway, in future we should probably offer this as optional behaviour. (this is related to #2616 proposal of an option to number footnotes per chaper, or section, even perhaps per page in PDF; here the topic is more of gathering them)

For this to be useful, as the footnotes will now be possibly far away from where text markers,
- the pdf should have backlinks to where from is the footnote referred to (possibly multiple times)
- the print-out on paper should perhaps have page indications of where from the footnote is referred to (like an index)

I don't know how to have pdf screen version only have a link, and printed version have a page indication, so already first goal would be to concentrate on the hyperlinking and have clickable backlinks.

**Describe the solution you'd like**
In short, have same functionalities in PDF as in html output for footnotes. Backrefs are useful, especially if footnotes are gathered not per page but at end of sectioning units.

Sphinx footnotes use footnotehyper based solutions to make footnotes real hyperlinks in more situations than feasible with vanilla LaTeX (in all situations now with with #8832). It would be good if the footnotehyper package itself got an upgrade to support
- backrefs from multiply referred-to footnotes,
- gathering footnotes per sectioning units,

so that the difficulties on our side are less.

Related: #2616, #8832
